visensor_calibration_flasher/src/visensor_calibration_flasher.cpp

```diff
@@ -112,63 +119,106 @@ int main(int argc, char **argv)
-    XmlRpc::XmlRpcValue intrinsics = cam_params["intrinsics"];
+    std::cout << "distortion_model is: " << distortion_model << std::endl;
+    if (distortion_model == std::string("radtan"))  {
+      visensor::ViCameraLensModelRadial::Ptr lens_model = camera_calibration.getLensModel<visensor::ViCameraLensModelRadial>();
+      std::cout << "lens_model radial and params are: " << distortion_coeffs << std::endl;
+      lens_model->setType();
+      lens_model->k1_ = (double) distortion_coeffs[0];
+      lens_model->k2_ = (double) distortion_coeffs[1];
+      lens_model->r1_ = (double) distortion_coeffs[2];
+      lens_model->r2_ = (double) distortion_coeffs[3];
+    }
+    else if (distortion_model == std::string("equidistant")) {
+      visensor::ViCameraLensModelEquidistant::Ptr lens_model = camera_calibration.getLensModel<visensor::ViCameraLensModelEquidistant>();
+      lens_model->setType();
+      lens_model->k1_ = (double) distortion_coeffs[0];
+      lens_model->k2_ = (double) distortion_coeffs[1];
+      lens_model->k3_ = (double) distortion_coeffs[2];
+      lens_model->k4_ = (double) distortion_coeffs[3];
+    }
+    else {
+      std::cout << "Distortion Model is not supported. Abort, abort!\n";
+      return 1;
+    }
 
-    camera_calibration.focal_point[0] = (double) intrinsics[0];
-    camera_calibration.focal_point[1] = (double) intrinsics[1];
-    camera_calibration.principal_point[0] = (double) intrinsics[2];
-    camera_calibration.principal_point[1] = (double) intrinsics[3];
+    std::cout << "lens model is: " << static_cast<int>(camera_calibration.lens_model_->type_) << std::endl;
+
+    camera_model.assign(cam_params["camera_model"]);
+    XmlRpc::XmlRpcValue intrinsics = cam_params["intrinsics"];
+    std::cout << "projection_model is: " << camera_model << std::endl;
+    if (camera_model == std::string("pinhole")) {
+      visensor::ViCameraProjectionModelPinhole::Ptr projection_model = camera_calibration.getProjectionModel<visensor::ViCameraProjectionModelPinhole>();
+      projection_model->setType();
+      projection_model->focal_length_u_ = (double) intrinsics[0];
+      projection_model->focal_length_v_ = (double) intrinsics[1];
+      projection_model->principal_point_u_ = (double) intrinsics[2];
+      projection_model->principal_point_v_ = (double) intrinsics[3];
+    }
+    else if (camera_model == std::string("omnidirectional")){
+      visensor::ViCameraProjectionModelOmnidirectional::Ptr projection_model = camera_calibration.getProjectionModel<visensor::ViCameraProjectionModelOmnidirectional>();
+      projection_model->setType();
+      projection_model->focal_length_u_ = (double) intrinsics[0];
+      projection_model->focal_length_v_ = (double) intrinsics[1];
+      projection_model->principal_point_u_ = (double) intrinsics[2];
+      projection_model->principal_point_v_ = (double) intrinsics[3];
+      // todo(lschmid) how to check if intrinsics has 5 attributes?
```

Comment:
Check out http://xmlrpcpp.sourceforge.net/doc/classXmlRpc_1_1XmlRpcValue.html#a30 . Should be what youre looking for, I think